packages/node/test/machine/integration/install-then-set-state.spec.ts

```diff
@@ -137,45 +144,68 @@ describe("Scenario: install AppInstance, set state, put on-chain", () => {
         gasLimit: SETSTATE_COMMITMENT_GAS
       });
 
+      const setStateCommitmentForFreeBalance = new SetStateCommitment(
+        network,
+        stateChannel.freeBalance.identity,
+        stateChannel.freeBalance.hashOfLatestState,
+        stateChannel.freeBalance.versionNumber,
+        stateChannel.freeBalance.timeout
+      );
+
+      await wallet.sendTransaction({
+        ...setStateCommitmentForFreeBalance.getSignedTransaction([
+          multisigOwnerKeys[0].signDigest(
+            setStateCommitmentForFreeBalance.hashToSign()
+          ),
+          multisigOwnerKeys[1].signDigest(
+            setStateCommitmentForFreeBalance.hashToSign()
+          )
+        ]),
+        gasLimit: SETSTATE_COMMITMENT_GAS
+      });
+
       // tslint:disable-next-line:prefer-array-literal
-      for (const _ of Array(appInstance.timeout)) {
+      for (const _ of Array(identityAppInstance.timeout)) {
         await provider.send("evm_mine", []);
       }
 
       await appRegistry.functions.setOutcome(
-        appInstance.identity,
-        appInstance.encodedLatestState
+        identityAppInstance.identity,
+        identityAppInstance.encodedLatestState
       );
 
-      const installCommitment = new InstallCommitment(
+      await appRegistry.functions.setOutcome(
+        stateChannel.freeBalance.identity,
+        stateChannel.freeBalance.encodedLatestState
+      );
+
+      const conditionalTransaction = new ConditionalTransaction(
         network,
         stateChannel.multisigAddress,
         stateChannel.multisigOwners,
-        appInstance.identity,
-        freeBalanceETH.identity,
-        freeBalanceETH.hashOfLatestState,
-        freeBalanceETH.versionNumber,
-        freeBalanceETH.timeout,
-        appInstance.appSeqNo,
-        network.ETHInterpreter,
+        identityAppInstance.identityHash,
+        stateChannel.freeBalance.identityHash,
+        network.CoinTransferETHInterpreter,
         defaultAbiCoder.encode(
-          ["uint256"],
-          [freeBalanceETH.coinTransferInterpreterParams!.limit]
+          ["tuple(uint256 limit)"],
```

Comment:
This encoding should be defined in one place and used throughout the rest of the codebase.